### ISSUE TITLE
Linting: Fix all instances of Ruff PT011 Error

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#7865]
### Provide a general description of the issue or problem.

Currently in our `.ruff.toml` file we ignore the PT011 rule:

https://github.com/sunpy/sunpy/blob/b656375f01eb2d2eceb0b1619c2a2e7851baa548/.ruff.toml#L28

This issue is to address all instances of this lint in the code and remove the ignore.

The Ruff error [PT011](https://docs.astral.sh/ruff/rules/pytest-raises-too-broad/) is triggered when in a test [`pytest.raises`](https://docs.pytest.org/en/4.6.x/reference.html#pytest-raises) is used without a `match=` keyword argument.

For example:

```python
def test_thing():
    pytest.raises(ValueError):
        thing()
```

This should be:

```python
def test_thing():
    pytest.raises(ValueError, match="thing should be called with an argument"):
        thing()
```

The `match=` keyword argument takes any regex which matches the error raised. When writing a match string it is best to pick either all or part of the error message which is unique and long enough to make sure the error being raised is specific to the code under test. If the error message is very long (>80 characters) it's best to not include the whole string. In some cases when writing a match string you might need to escape special characters or pick a substring without them.

To address this issue the following steps should be performed:

1. Remove the line from .ruff.toml
2. run `tox -e codestyle` which will give you a list of errors
3. For each error, you need to go through and add a `match=` kwarg specific to that error being raised in that test. One approach to this is to add a match kwarg which will not match i.e. `match="!!"` which will cause the test to fail when run with `tox -e py312` and the test fail will show you the error that was raised and you can pick a suitable string to put in that match kwarg.
4. Ensure that `tox -e codestyle` passes with no errors once all issues are addressed.
5. Open a pull request.

**Comment by raghav20232023** [comment]
Hi @Cadair 
I have completed the initial step of identifying the errors (total 70 found) and have made changes as follows :
added the "match: kwarg and set its value to "!!"

As mentioned by you, this will help identify the exception better, which will help in identifying the right strings to be set as value of match kwarg for each of 70 errors individually. 

My next step here will be to do the above. 

I have created a pull request #7866 
link : https://github.com/sunpy/sunpy/pull/7866

Kindly suggest me any further steps that need to be taken. Your input is highly appreciated.

Regards
